src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java

```diff
@@ -1,50 +1,117 @@
+@SuppressWarnings({"squid:ClassVariableVisibilityCheck", "squid:S2444"})
+public class ModBlocks
 {
     /*
      * Creating objects for all blocks in the mod.
      * References can be made to here.
      */
-    public static final Block blockHutTownHall            = new BlockHutTownHall();
-    public static final Block blockHutCitizen             = new BlockHutCitizen();
-    public static final Block blockHutMiner               = new BlockHutMiner();
-    public static final Block blockHutLumberjack          = new BlockHutLumberjack();
-    public static final Block blockHutBaker               = new BlockHutBaker();
-    public static final Block blockHutBuilder             = new BlockHutBuilder();
-    public static final Block blockHutDeliveryman         = new BlockHutDeliveryman();
-    public static final Block blockHutBlacksmith          = new BlockHutBlacksmith();
-    public static final Block blockHutStonemason          = new BlockHutStonemason();
-    public static final Block blockHutFarmer              = new BlockHutFarmer();
-    public static final Block blockHutFisherman           = new BlockHutFisherman();
-    public static final Block blockSubstitution           = new BlockSubstitution();
-    public static final Block blockSolidSubstitution      = new BlockSolidSubstitution();
-    public static final Block blockHutField               = new BlockHutField();
-    public static final Block blockHutGuardTower          = new BlockHutGuardTower();
-    public static final Block blockHutWareHouse           = new BlockHutWareHouse();
-    public static final Block blockConstructionTape       = new BlockConstructionTape();
-    public static final Block blockConstructionTapeCorner = new BlockConstructionTapeCorner();
-    public static final Block blockTimberFrame            = new BlockTimberFrame();
-    public static final Block blockRack                   = new BlockMinecoloniesRack();
-    public static final Block blockWayPoint               = new BlockWaypoint();
-    public static final Block blockHutBarracks            = new BlockHutBarracks();
-    public static final Block blockHutBarracksTower       = new BlockHutBarracksTower();
-    public static final Block blockInfoPoster             = new BlockInfoPoster();
-
+    public static BlockHutTownHall            blockHutTownHall;
+    public static BlockHutCitizen             blockHutCitizen;
+    public static BlockHutMiner               blockHutMiner;
+    public static BlockHutLumberjack          blockHutLumberjack;
+    public static BlockHutBaker               blockHutBaker;
+    public static BlockHutBuilder             blockHutBuilder;
+    public static BlockHutDeliveryman         blockHutDeliveryman;
+    public static BlockHutBlacksmith          blockHutBlacksmith;
+    public static BlockHutStonemason          blockHutStonemason;
+    public static BlockHutFarmer              blockHutFarmer;
+    public static BlockHutFisherman           blockHutFisherman;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this "public static blockSubstitution" field final [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1444)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make blockSubstitution a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)